manager/controllers/package_detail.go

```diff
@@ -0,0 +1,124 @@
+package controllers
+
+import (
+	"app/base/database"
+	"app/base/models"
+	"app/base/utils"
+	"github.com/gin-gonic/gin"
+	"github.com/jinzhu/gorm"
+	"net/http"
+)
+
+func singlePackageLatestQuery(pkgName string) *gorm.DB {
+	return database.Db.Table("package pkg").Select("*").Joins("JOIN package_name pn ON pkg.name_id = pn.id").
+		Joins("JOIN advisory_metadata adv ON pkg.advisory_id = adv.id").
+		Where("pn.name = ?", pkgName).
+		Order("adv.public_date DESC").Limit(1)
+}
+
+func singlePackageEvraQuery(pkgName string, evra string) *gorm.DB {
+	return database.Db.Table("package").
+		Joins("JOIN package_name ON package.name_id = package_name.id").
+		Where("package_name.name = ?", pkgName).
+		Where("package.evra = ?", evra)
+}
+
+func getAdvisoryByID(advID int) (models.AdvisoryMetadata, error) {
+	var advisory models.AdvisoryMetadata
+	err := database.Db.Where("id = ?", advID).First(&advisory).Error
+	return advisory, err
+}
+
+type PackageDetailResponse struct {
```

Comment:
Please make the structures consistent with AdvisoryDetailResponse and SystemDetailResponse. It's still different.
~~~go
type PackageDetailResponse struct {
	Data PackageDetailItem `json:"data"`
}

type PackageDetailItem struct {
	Attributes PackageDetailAttributes  `json:"attributes"`
	ID         string                   `json:"id"` // package nevra
	Type       string                   `json:"type"` // this will be always "package"
}

type PackageDetailAttributes struct {
	Description  string   `json:"description"` // not hashes for des. and sum. but content itself
	Summary      string   `json:"summary"`
	Name         string   `json:"name"` // package name
	Version      string   `json:"version"` // put whole evra here for now
	AdvisoryID   string   `json:"advisory_id"` // e.g. RHBA-2020:1234
	// it seems that's all we have now, maybe we will split version and arch later
}
~~~
@semtexzv @MichaelMraka please, correct me if I am wrong.